anilistUpdater/main.lua

```diff
@@ -98,57 +158,54 @@ if not conf_path then
     mp.msg.warn("Could not find or create anilistUpdater.conf in any known location! Using default options.")
 end
 
--- Now load options as usual
-local options = {
-    DIRECTORIES = "",
-    EXCLUDED_DIRECTORIES = "",
-    UPDATE_PERCENTAGE = 85,
-    SET_COMPLETED_TO_REWATCHING_ON_FIRST_EPISODE = false,
-    UPDATE_PROGRESS_WHEN_REWATCHING = true,
-    SET_TO_COMPLETED_AFTER_LAST_EPISODE_CURRENT = true,
-    SET_TO_COMPLETED_AFTER_LAST_EPISODE_REWATCHING = true
-}
+-- Initialize options from default_options
+local options = {}
+for _, option in ipairs(default_options) do
+    options[option.key] = option.value
+end
 if conf_path then
+    -- Read the current config file content
+    local current_config = ""
+    local config_file = io.open(conf_path, "r")
+    if config_file then
+        current_config = config_file:read("*all")
+        config_file:close()
+    end
+    
+    -- This will override the defaults with values from the config file
     mpoptions.read_options(options, "anilistUpdater")
-end
-
-local function normalize_path(p)
-    p = p:gsub("\\", "/")
-    if p:sub(-1) == "/" then
-        p = p:sub(1, -2)
+    
+    -- Check for missing options and append them
+    local missing_options = {}
+    for _, option in ipairs(default_options) do
+        if not current_config:find(option.key .. "=") then
+            table.insert(missing_options, option.key .. "=" .. option.config_value)
+        end
     end
-    return p
-end
-
--- Parse DIRECTORIES if it's a string (comma or semicolon separated)
-if type(options.DIRECTORIES) == "string" and options.DIRECTORIES ~= "" then
-    local dirs = {}
-    for dir in string.gmatch(options.DIRECTORIES, "([^,;]+)") do
-        local trimmed = (dir:gsub("^%s*(.-)%s*$", "%1"):gsub('[\'"]', '')) -- trim
-        table.insert(dirs, normalize_path(trimmed))
+    
+    -- Append missing options
+    if #missing_options > 0 then
+        local append_file = io.open(conf_path, "a")
+        if append_file then
+            for _, option in ipairs(missing_options) do
+                append_file:write(option .. "\n")
+            end
+            append_file:close()
+        end
```

Comment:
I don't think we should check and append missing options. When the .conf file is generated it includes every option, if the user wants to delete some that they dont use they should be able to

If we remove this then theres also probably no reason to have `default_options`, directly load options like before instead of creating it from default_options